src/Paprika/Store/RootPage.cs

```diff
@@ -225,12 +230,49 @@ public void Destroy(IBatchContext batch, in NibblePath account)
 
         // Remove the cached
         batch.IdCache.Remove(account.UnsafeAsKeccak);
+    }
 
-        // TODO: there' no garbage collection for storage
-        // It should not be hard. Walk down by the mapped path, then remove all the pages underneath.
+    public void DeleteByPrefix(in Key prefix, IBatchContext batch)
+    {
+        if (prefix.IsState)
+        {
+            var data = batch.TryGetPageAlloc(ref Data.StateRoot, PageType.Standard);
+            var updated = new Merkle.StateRootPage(data).DeleteByPrefix(prefix.Path, batch);
+            Data.StateRoot = batch.GetAddress(updated);
+        }
+        else
+        {
+            scoped NibblePath id;
+            Span<byte> idSpan = stackalloc byte[sizeof(uint)];
+
+            var keccak = prefix.Path.UnsafeAsKeccak;
```

Comment:
Is it ok to use `UnsafeAsKeccak` here for paths that will be < 64 ?

Reply:
In line 237, there's a `.IsState` check. If the path is shorter than 64 or is 64 and is account, it's a state tree. It'll never bee less than 64 for the storage. The Merkle, the root of a storage tree will have the path set to 64.